torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -8820,4 +8821,108 @@ def celu_hook(module, _input, output):
+            map_location = {'cuda:%d' % 0: 'cuda:%d' % rank}
+            checkpoint = torch.load(chkpt_file, map_location=map_location)
+
+            dummy_ddp_model.load_state_dict(checkpoint['state_dict'])
+            dummy_hook = pickle.loads(checkpoint['comm_hook'])
+
+            with self.assertLogs() as captured:
+                dummy_hook_state = pickle.loads(checkpoint['comm_hook_state'])
+
+            # Check that the logger has only one entry
+            self.assertEqual(len(captured.records), 1)
+            # Check that the logger has an expected entry
+            self.assertEqual(
+                captured.records[0].getMessage(),
+                "NOTE: PowerSGDState is set to use the default process group."
+            )
+            # Check that loaded function is correct
+            self.assertEqual(dummy_hook.__qualname__, hook.__qualname__)
+
+            slots_original = hook_state.__slots__
+            slots_restored = dummy_hook_state.__slots__
+            self.assertEqual(slots_original, slots_restored)
+
+            slots = copy.copy(slots_restored)
+            slots.remove("process_group")
+            slots.remove("rng")
+
+            # Check that all attributes are restored correctly.
+            # Excluding ``process_group`` and ``rng``.
+            for entry in slots:
+                self.assertEqual(getattr(dummy_hook_state, entry), getattr(hook_state, entry))
+
+            # Check that ``process_group`` was set to default
+            self.assertEqual(dummy_hook_state.process_group, _get_default_group())
+
+            # Check that a random state was restored properly:
+            # :met:`np.random.RandomState.get_state` returns a tuple with entries:
+            # `bit_generator` - str,
+            # `state.key` - ndarray dtype[uint32],
+            # `state.pos` - int,
+            # `has_gauss` - int,
+            # `gauss` - float
+            #  (refer to https://github.com/numpy/numpy/blob/266aad7478bc7fbcc55eea7f942a0d373b838396/numpy/random/mtrand.pyi)
+            # To make sure random state was restored properly, all entries should equal the original
+            for entry1, entry2 in zip(hook_state.rng.get_state(), dummy_hook_state.rng.get_state()):
+                np.testing.assert_array_equal(entry1, entry2)
+
+            if rank == 0:
+                os.remove(chkpt_file)
+
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["cuda"],
+            f"The {BACKEND} backend does not support DDP communication hook on CUDA devices"
+        )
+        @skip_if_lt_x_gpu(int(os.environ["WORLD_SIZE"]))
+        def test_ddp_hook_pickling_powerSGD(self):
+
+            hook = powerSGD.powerSGD_hook
+            powersgd_state = powerSGD.PowerSGDState(
+                process_group=None,
+                matrix_approximation_rank=1,
+                start_powerSGD_iter=4,
+            )
+            self._test_hook_pickling(hook, powersgd_state)
+
+
```

Comment:
nit: I think we only need to leave two newlines between the end of he test and the `instantiate_parametrized_tests()`,

Reply:
Noted, thanks!